File: tape_equilibrium.py
def solution(A):
    if len(A) == 2:
        return abs(A[1]-A[0])
        
    sums_left = {}
    sums_right = {}
    current_left_sum = A[0]
    current_right_sum = A[-1]
    
    for index in range(1, len(A)):
        sums_left[index] = current_left_sum
        sums_right[len(A)-index] = current_right_sum
        current_left_sum += A[index]
        current_right_sum += A[len(A)-index-1]
    
    differences = []
    
    for index in range(1, len(A)):
        differences.append(abs(sums_left[index] - sums_right[index]))
        
    return  min(differences);

File: TapeEquilibrium.cs
using System;
using System.Collections.Generic;
using System.Linq;

class Solution {
    public int solution(int[] A) {
        if (A.Length == 2)
            return Math.Abs(A.First() - A.Last());
        
        Dictionary<int,int> sumsLeft = new Dictionary<int,int>();
        Dictionary<int,int> sumsRight = new Dictionary<int,int>();
        int currentLeftSum = A.First();
        int currentRightSum = A.Last();
        
        for (int index = 1; index < A.Length; index++)
        {
            sumsLeft.Add(index, currentLeftSum);   
            sumsRight.Add(A.Length-index, currentRightSum);           
            currentLeftSum += A[index];
            currentRightSum += A[A.Length-index-1];
        }
        
        List<int> differences = new List<int>();;
        
        for(int index = 1; index < sumsLeft.Count; index++)
        {
            differences.Add(Math.Abs(sumsLeft[index] - sumsRight[index]));
        }
        
        return  differences.Min();
    }
}